src/components/VideoChatMenu.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Popover from './Popover';
+import withWindowDimensions, {windowDimensionsPropTypes} from './withWindowDimensions';
+import MenuItem from './MenuItem';
+import openURLInNewTab from '../libs/openURLInNewTab';
+import ZoomIcon from '../../assets/images/zoom-icon.svg';
+import GoogleMeetIcon from '../../assets/images/google-meet.svg';
+import CONST from '../CONST';
+
+const propTypes = {
+    // State that determines whether to display the create menu or not
+    isVisible: PropTypes.bool.isRequired,
+
+    // Callback that determines behavior when menu is closed
+    onClose: PropTypes.func.isRequired,
+
+    ...windowDimensionsPropTypes,
+};
+
+const VideoChatMenu = (props) => {
+    const menuItemData = [
```

Comment:
So a side-effect of declaring this in the render function is that it's redeclared every time this component renders, which is maybe not a big deal, but also not so good for performance. I think it would be better to change this to be a class component, then declare `this.menuItemData` in the constructor. That way, you can still access props in the array but it's only declared once.

Reply:
I think I disagree here. It says in our [React style guide](https://github.com/Expensify/Style-Guide/blob/master/react.md#stateless-components-vs-pure-components-vs-class-based-components-vs-render-props---when-to-use-what) that we should use class components when we need to maintain state or use lifecycle methods, neither of which are happening here. Also this menu gets opened once 90% of the time.

Reply:
Okay, well I _think_ the `constructor` counts as a lifecycle method, but I'll downgrade this to NAB if that's what you think is best. Redeclaring the array with every render most likely won't have a significant performance impact.